include/envoy/network/connection.h

```diff
@@ -69,6 +69,47 @@ enum class ConnectionCloseType {
                      // until the delayed_close_timeout expires
 };
 
+/**
+ * Run-time modifiable connection options to alter socket settings
+ */
+class DynamicSocketOptions {
+public:
+  virtual ~DynamicSocketOptions() = default;
+
+  /**
+   * Set the receive buffer size for the socket to a value specified by the 'buffer_size'
+   * @param buffer_size the size of receive buffer intended to be modified to
+   * @return Api::SysCallIntResult an instance consisting of return code and error number
+   *         for executing the system call
+   */
+  virtual Api::SysCallIntResult setSocketRecvBufferSize(int buffer_size) const PURE;
```

Comment:
Please use unsigned explicit types for function signatures, such as uint32_t, etc. Same below.

Reply:
@mattklein123 I initially made an implementation with 'uint32_t'. Actually @alyssawilk suggested I change this to 'int', which I seconded after some contemplation.

Perhaps, explicit types might not be supported on certain machines.

WDYT ?

From a learning perspective, I want to understand the actual reason for such strictness.

Thanks!

Reply:
Because buffer sizes cannot be negative. I don't think we should worry about machine/OS specific platform APIs at this layer. If Linux uses ints for this that is fine and the mapping can be done in the implementation.

Reply:
@alyssawilk the underlying call `virtual SysCallIntResult setsockopt(os_fd_t sockfd, int level, int optname, const void* optval, socklen_t optlen) PURE;`  and linux also take const void* optval.

I can make the change back to uint32_t, please let me know if you have other opinions.

Thanks!

Reply:
Please switch this to uint32_t. It can stay int at the syscall layer.